submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionStatsDataRepository.kt

```diff
@@ -17,37 +15,31 @@ import org.springframework.data.mongodb.core.query.Update.update
 
 class SubmissionStatsDataRepository(
     private val mongoTemplate: MongoTemplate,
-    private val submissionRepository: SubmissionMongoRepository
-) : SubmissionMongoRepository by submissionRepository {
-    fun updateStat(accNo: String, version: Int, stat: SubmissionStat) {
-        val query = findStatsQuery(accNo, version, stat.type.name)
-        mongoTemplate.updateFirst(query, update("$SUB_STATS.$.$STAT_DOC_VALUE", stat.value), DocSubmission::class.java)
+    private val statsRepository: SubmissionStatsRepository,
+) : SubmissionStatsRepository by statsRepository {
+    fun updateOrRegisterStat(stat: SubmissionStat) {
+        val query = findStatsQuery(stat.accNo)
+        mongoTemplate.upsert(query, update("$SUB_STATS.${stat.type}", stat.value), DocSubmissionStats::class.java)
     }
 
-    fun incrementStat(accNo: String, version: Int, stats: List<SubmissionStat>): Long {
+    fun incrementStat(accNo: String, stats: List<SubmissionStat>): Long {
         var increment = 0L
-        val bulk = mongoTemplate.bulkOps(UNORDERED, DocSubmission::class.java)
+        val bulk = mongoTemplate.bulkOps(UNORDERED, DocSubmissionStats::class.java)
 
         stats.forEach {
-            bulk.incrementStat(accNo, version, it)
+            bulk.incrementStat(accNo, it)
             increment += it.value
         }
 
         bulk.execute()
         return increment
     }
 
-    private fun findStatsQuery(accNo: String, version: Int, name: String) =
-        Query(
-            where(DocSubmissionFields.SUB_ACC_NO).`is`(accNo).andOperator(
-                where(DocSubmissionFields.SUB_VERSION).`is`(version),
-                where("$SUB_STATS.$STAT_DOC_NAME").`is`(name)
-            )
-        )
+    private fun findStatsQuery(accNo: String) = Query(where(DocSubmissionFields.SUB_ACC_NO).`is`(accNo))
```

Comment:
do not see benefics in this inline, I think code is simpler adding expression everywhere (one does not need to check this method to see what is happening)